### PR TITLE
compile test

### PR DESCRIPTION
**Pull request by step21**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Adding compile test, as this issue https://github.com/conda-forge/ghc-feedstock/issues/9 still exists, so at least on some architectures the compiler is broken.

### CODE REVIEW

File: .scripts/build_steps.sh [hunk 1]
```diff
@@ -28,7 +28,6 @@ CONDARC
 BUILD_CMD=build
 
 conda install --yes --quiet "conda-forge-ci-setup=3" conda-build pip ${GET_BOA:-} -c conda-forge
-conda update --yes -c conda-forge/label/lief_dev py-lief
```
Comment (jakirkham) [inline .scripts/build_steps.sh:31]: Think this is still needed

File: .scripts/run_osx_build.sh [hunk 2]
```diff
@@ -24,7 +24,6 @@ conda activate base
 
 echo -e "\n\nInstalling conda-forge-ci-setup=3 and conda-build."
 conda install -n base --quiet --yes "conda-forge-ci-setup=3" conda-build pip ${GET_BOA:-}
-conda update --yes -c conda-forge/label/lief_dev py-lief
```
Comment (jakirkham) [inline .scripts/run_osx_build.sh:27]: Also this

**Review by jakirkham — COMMENTED** [review]
Looks like re-rendering removed a couple things that Uwe added. Think we still need these for the build to work. Have indicated them below

File: .scripts/build_steps.sh (hunk 1 above)
Comment (step21) [reply .scripts/build_steps.sh:31]: oops. sry, I didn't realize there was custom stuff. (though I looked at the PR)

Comment (jakirkham) [reply .scripts/build_steps.sh:31]: Not at all. Just trying to save you from debugging things that were fixed by this change 🙂

Comment (step21) [reply .scripts/build_steps.sh:31]: Thanks. That is very much appreciated :)

**Pull request merged**